# Goal Split Subgroup: L2 Escape Model

The following models takes into account that, throughout analyses, *no_goal* seems to behave entirely differently to the other two Goal Type conditions (*goal_frequent* and *goal_non_frequent*). To do so, it splits the data accordingly and looks into goals on the one hand and no_goals on the other separately. 

Import Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Goal Split Subgroup

1. Create escape_L2 (goals and no_goals)
2. Counts and Proportions (goals and no_goals)
3. Logistic Models

    - Goals

        Main Effect Models:

        - E01: Focus
        - E02: Agent
        - E03: Goal_Type

        Additive Models:

        - E04: Focus + Agent
        - E05: Focus + Goal_Type
        - E06: Agent + Goal_Type

        Interactive Models:

        - E07: Focus * Agent
        - E08: Focus * Goal_Type
        - E09: Agent * Goal_Type
        - E10: Focus * Agent + Goal_Type

    - No_goals

        Main Effect Models:

        - E11:
        - E12:

        Additive Model:

        - E13:

        Interactive Model:

        - E14:

4. Model Comparisons
5. Summary Table

Read Data

In [2]:
subgroup_theoretical = pd.read_csv("../../data/processed/subgroup_theoretical.csv")

1. Create escape_2

Binary Outcome: responses that are not "L2_other": those who managed to escape L2_other by choosing either a *correct* response or an *L1_transfer* response.

In [3]:
subgroup_theoretical["escape_L2"] = (
    subgroup_theoretical["Response_Full"] != "L2_other"
).astype(int)

Split data into goals vs. no_goals

In [4]:
goals = subgroup_theoretical[subgroup_theoretical["Goal_Type"]!= "no_goal"].copy()

In [5]:
no_goal = subgroup_theoretical[subgroup_theoretical["Goal_Type"]== "no_goal"].copy()

2. Counts and Proportions for goals and no_goal.

General escape_L2 reponses count:

In [6]:
escape_L2_counts = pd.crosstab(
    subgroup_theoretical["escape_L2"],
    subgroup_theoretical["Response_Full"],
    margins= True)

escape_L2_counts

Response_Full,L1_transfer,L2_other,correct,missing_response,All
escape_L2,,,,,
0,0,38,0,0,38
1,96,0,61,3,160
All,96,38,61,3,198


Escape_L2 responses count for goals:

In [7]:
goal_escape_L2_counts = pd.crosstab(
    goals["escape_L2"],
    goals["Response_Full"],
    margins= True)

goal_escape_L2_counts

Response_Full,L1_transfer,L2_other,correct,missing_response,All
escape_L2,,,,,
0,0,26,0,0,26
1,62,0,43,1,106
All,62,26,43,1,132


Escape_L2 responses count for no_goals:

In [8]:
no_goal_escape_L2_counts = pd.crosstab(
    no_goal["escape_L2"],
    no_goal["Response_Full"],
    margins= True)

no_goal_escape_L2_counts

Response_Full,L1_transfer,L2_other,correct,missing_response,All
escape_L2,,,,,
0,0,12,0,0,12
1,34,0,18,2,54
All,34,12,18,2,66


General escape_L2 responses proportions:

In [9]:
escape_L2_props = subgroup_theoretical["escape_L2"].value_counts(normalize = True)
escape_L2_props

escape_L2
1    0.808081
0    0.191919
Name: proportion, dtype: float64

From 198 observations, 80.8% of responses managed to escape L2_other options. 

Escape_L2 responses proportions for goals:

In [10]:
goals_escape_L2_props = goals["escape_L2"].value_counts(normalize = True)
goals_escape_L2_props

escape_L2
1    0.80303
0    0.19697
Name: proportion, dtype: float64

From 132 observations in the goal group, 80.3% of responses managed to escape L2_other options. 

Escape_L2 responses proportions for no_goals:

In [11]:
no_goal_escape_L2_props = no_goal["escape_L2"].value_counts(normalize = True)
no_goal_escape_L2_props

escape_L2
1    0.818182
0    0.181818
Name: proportion, dtype: float64

From 66 observations in the no_goal group, 81.8% of responses managed to escape L2_other options. 

Counts and Proportions by Condition:

In [12]:
goals_counts_condition = pd.crosstab(
    [goals["Goal_Type"], goals["Agent"], goals["Focus"]],
    goals["escape_L2"],
    margins = True)

goals_counts_condition

escape_L2                       0    1  All
Goal_Type         Agent Focus              
goal_frequent     0     I       8   16   24
                        They    6   12   18
                  1     I       2   14   16
                        They    0    8    8
goal_non_frequent 0     I       8   16   24
                        They    1   17   18
                  1     I       1   15   16
                        They    0    8    8
All                            26  106  132

In [13]:
no_goal_counts_condition = pd.crosstab(
    [no_goal["Agent"],no_goal["Focus"]],
    no_goal["escape_L2"],
    margins = True)

no_goal_counts_condition

escape_L2     0   1  All
Agent Focus             
0     I       3  21   24
      They    5  13   18
1     I       2  14   16
      They    2   6    8
All          12  54   66

In [14]:
goal_props_condition = pd.crosstab(
    [goals["Goal_Type"], goals["Agent"], goals["Focus"]],
    goals["escape_L2"],
    normalize= "index")

goal_props_condition

escape_L2                             0         1
Goal_Type         Agent Focus                    
goal_frequent     0     I      0.333333  0.666667
                        They   0.333333  0.666667
                  1     I      0.125000  0.875000
                        They   0.000000  1.000000
goal_non_frequent 0     I      0.333333  0.666667
                        They   0.055556  0.944444
                  1     I      0.062500  0.937500
                        They   0.000000  1.000000

Within goals, Agency seems to assist in the process of escaping the L2_other options.

As previously mentioned, the condition with the strongest L1 transfer (goal_non_frequent, agent=1) exhibits the highest L2 escape rate (95.8%). This suggests that failure does not occur at the L2 suppression stage, but at the subsequent resolution stage.

In [15]:
no_goal_props_condition = pd.crosstab(
    [no_goal["Agent"],no_goal["Focus"]],
    no_goal["escape_L2"],
    normalize= "index")

no_goal_props_condition

escape_L2           0         1
Agent Focus                    
0     I      0.125000  0.875000
      They   0.277778  0.722222
1     I      0.125000  0.875000
      They   0.250000  0.750000

Within no_goals, however, Agency doesn't seem to play the same part in the process of escaping the L2_other options compared to the goal group.

Agency appears to help participants escape L2. 

As previously mentioned, if there is a goal representation available, agency seems to help participants commit by leaving the L2 space; however, if there is no goal representation, agency seems to have much less to work with.

3. Logistic Models

Import Libraries

In [16]:
import statsmodels.formula.api as smf
import statsmodels.api as sm
from scipy.stats import chi2


Sanity checks:

In [17]:
goals["escape_L2"].value_counts()

escape_L2
1    106
0     26
Name: count, dtype: int64

In [18]:
no_goal["escape_L2"].value_counts()

escape_L2
1    54
0    12
Name: count, dtype: int64

**Goals**

**Main Effect Models**

**E01**

escape_L2 ~ Focus

In [19]:
E01 = smf.logit(
    "escape_L2 ~ Focus",
    data=goals
    ).fit()

print(E01.summary())

Optimization terminated successfully.
         Current function value: 0.487863
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      130
Method:                           MLE   Df Model:                            1
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.01675
Time:                        16:41:07   Log-Likelihood:                -64.398
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                    0.1386
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         1.1664      0.263      4.440      0.000       0.652       1.681
Focus[T.They]     0.

Focus on its own isn't enough to predict anything again -although it is way better on its own once there is a goal split in the data.

**E02**

escape_L2 ~ Agent

In [20]:
E02 = smf.logit(
    "escape_L2 ~ Agent",
    data=goals
    ).fit()

print(E02.summary())

Optimization terminated successfully.
         Current function value: 0.458568
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      130
Method:                           MLE   Df Model:                            1
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.07579
Time:                        16:41:07   Log-Likelihood:                -60.531
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.001628
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.9754      0.245      3.986      0.000       0.496       1.455
Agent          1.7327      0.

When the data was combined, agency on its own's p-value = 0.013 (coeff = 1.1086). Once the data is the goal split is introduced, agency coefficient increases (1.7327) and p-value improves ( 0.007). Thus, after removing 66 observations, the effect gets actually stronger. 

**E03**

escape_L2 ~ Goal_Type

In [21]:
E03 = smf.logit(
    "escape_L2 ~ Goal_Type",
    data=goals
    ).fit()

print(E03.summary())

Optimization terminated successfully.
         Current function value: 0.489593
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      130
Method:                           MLE   Df Model:                            1
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.01326
Time:                        16:41:07   Log-Likelihood:                -64.626
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                    0.1875
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          1.1394      0.287      3.967     

Goal Type still contributes nothing on its own. But, once again, working with the goal split data has improved the model's LLR p-value (0.1875) compared to combined data (0.4060)

**Additive Models**

**E04**

escape_L2 ~ Focus + Agent

In [22]:
E04 = smf.logit(
    "escape_L2 ~  Focus + Agent",
    data=goals
    ).fit()

print(E04.summary())

Optimization terminated successfully.
         Current function value: 0.446122
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      129
Method:                           MLE   Df Model:                            2
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                  0.1009
Time:                        16:41:07   Log-Likelihood:                -58.888
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.001351
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.6432      0.298      2.156      0.031       0.058       1.228
Focus[T.They]     0.

Focus on its own couldn't predict anything. Again, once it is added to the model with agency, while it is still non-significant (p-value 0 0.080), it is much more informative than boefore. The coefficient also grows from 0.694 to 0.876; so, once agency is accounted for, focus explains additional variation in the data.

The whole model's LLR p-value (0.00135) is also better than agent alone (0.00163); not a lot but not requiring an interaction.

This model explains a lot of variation and demostrates parsimony.


**E05**

escape_L2 ~ Focus + Goal_Type

In [23]:
E05 = smf.logit(
    "escape_L2 ~  Focus + Goal_Type",
    data=goals
    ).fit()

print(E05.summary())

Optimization terminated successfully.
         Current function value: 0.481177
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      129
Method:                           MLE   Df Model:                            2
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.03022
Time:                        16:41:07   Log-Likelihood:                -63.515
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                    0.1381
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.8931      0.326      2.738     

Goal Type works better with Focus that on its own but the model is still very poor. 

**E06**

escape_L2 ~ Agent + Goal_Type

In [24]:
E06= smf.logit(
    "escape_L2 ~  Agent + Goal_Type",
    data=goals
    ).fit()

print(E06.summary())

Optimization terminated successfully.
         Current function value: 0.451528
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      129
Method:                           MLE   Df Model:                            2
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.08998
Time:                        16:41:07   Log-Likelihood:                -59.602
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.002758
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.6854      0.317      2.162     

This is a good model even though the goal_non_frequent's p is still not siginificant. It is contributing to the model. Given this, a full additive model that was not considered in the beginning should be looked into.

**P01**

escape_L2 ~ Focus + Agent + Goal_Type

In [25]:
P01= smf.logit(
    "escape_L2 ~  Focus + Agent + Goal_Type",
    data=goals
    ).fit()

print(P01.summary())

Optimization terminated successfully.
         Current function value: 0.438864
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      128
Method:                           MLE   Df Model:                            3
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                  0.1155
Time:                        16:41:07   Log-Likelihood:                -57.930
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.001709
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.3380      0.368      0.918     

Goal Type's p-value improves a little bit and the coefficient increases. The whole model is better (LLR p-value: 0.001709 vs. 0.002758) although not better than the focus-agency additive model's p (0.001351). 

Yet, this model shows a better picture of how counterfactual processing is shaped by considering the 3 cognitive dimensions (Focus, Agency and Goal_Type). In this sense, Agency shows a strong contribution, Focus shows a consistent positive contribution approaching significance and Goal Type shows a modest positive contribution. 

**Interactive Models**

**E07**

escape_L2 ~ Focus * Agent

In [26]:
E07 = smf.logit(
    "escape_L2 ~  Focus * Agent",
    data=goals
    ).fit()

print(E07.summary())

         Current function value: 0.441231
         Iterations: 35
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      128
Method:                           MLE   Df Model:                            3
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                  0.1107
Time:                        16:41:07   Log-Likelihood:                -58.243
converged:                      False   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.002293
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.6931      0.306      2.264      0.024       0.093       1.293
Focus[T.They]           0.7282      0.

/opt/anaconda3/envs/counterfactual-env/lib/python3.12/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


This gigantic interaction shows that nothing can be estimated reliably. Either one or more celss have very few observations. Yet, it support that there isn't enough evidence that the effect of agency depend on focus.

**E08**

escape_L2 ~ Focus * Goal_Type

In [27]:
E08 = smf.logit(
    "escape_L2 ~  Focus * Goal_Type",
    data=goals
    ).fit()

print(E08.summary())

Optimization terminated successfully.
         Current function value: 0.470484
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      128
Method:                           MLE   Df Model:                            3
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.05177
Time:                        16:41:07   Log-Likelihood:                -62.104
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                   0.07919
                                                   coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------------------
Intercept                               

**E09**

escape_L2 ~ Agent * Goal_Type

In [28]:
E09 = smf.logit(
    "escape_L2 ~  Agent * Goal_Type",
    data=goals
    ).fit()

print(E09.summary())

Optimization terminated successfully.
         Current function value: 0.451492
         Iterations 7
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      128
Method:                           MLE   Df Model:                            3
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.09005
Time:                        16:41:07   Log-Likelihood:                -59.597
converged:                       True   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.008117
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                                0.6931      0.3

**E10**

escape_L2 ~ Focus * Agent + Goal_Type

In [29]:
E10 = smf.logit(
    "escape_L2 ~ Focus * Agent + Goal_Type",
    data=goals
    ).fit()

print(E10.summary())

         Current function value: 0.434009
         Iterations: 35
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                  132
Model:                          Logit   Df Residuals:                      127
Method:                           MLE   Df Model:                            4
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                  0.1253
Time:                        16:41:07   Log-Likelihood:                -57.289
converged:                      False   LL-Null:                       -65.495
Covariance Type:            nonrobust   LLR p-value:                  0.002514
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.3901      0.374      1.044      0.297      -0.342       1.123
Focus

/opt/anaconda3/envs/counterfactual-env/lib/python3.12/site-packages/statsmodels/base/model.py:607: ConvergenceWarning: Maximum Likelihood optimization failed to converge. Check mle_retvals
  warnings.warn("Maximum Likelihood optimization failed to "


**No Goals**

**Main Effect Models**

**E11**

escape_L2 ~ Focus

In [30]:
E11 = smf.logit(
    "escape_L2 ~ Focus",
    data=no_goal
    ).fit()

print(E11.summary())

Optimization terminated successfully.
         Current function value: 0.457812
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                   66
Model:                          Logit   Df Residuals:                       64
Method:                           MLE   Df Model:                            1
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.03444
Time:                        16:41:07   Log-Likelihood:                -30.216
converged:                       True   LL-Null:                       -31.293
Covariance Type:            nonrobust   LLR p-value:                    0.1421
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         1.9459      0.478      4.070      0.000       1.009       2.883
Focus[T.They]    -0.

**E12**

escape_L2 ~ Agent

In [31]:
E12 = smf.logit(
    "escape_L2 ~ Agent",
    data=no_goal
    ).fit()

print(E12.summary())

Optimization terminated successfully.
         Current function value: 0.473694
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                   66
Model:                          Logit   Df Residuals:                       64
Method:                           MLE   Df Model:                            1
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:               0.0009392
Time:                        16:41:07   Log-Likelihood:                -31.264
converged:                       True   LL-Null:                       -31.293
Covariance Type:            nonrobust   LLR p-value:                    0.8084
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.4469      0.393      3.682      0.000       0.677       2.217
Agent          0.1625      0.

**Additive Model**

**E13**

escape_L2 ~ Focus + Agent

In [32]:
E13 = smf.logit(
    "escape_L2 ~ Focus + Agent",
    data=no_goal
    ).fit()

print(E13.summary())

Optimization terminated successfully.
         Current function value: 0.457728
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                   66
Model:                          Logit   Df Residuals:                       63
Method:                           MLE   Df Model:                            2
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.03461
Time:                        16:41:07   Log-Likelihood:                -30.210
converged:                       True   LL-Null:                       -31.293
Covariance Type:            nonrobust   LLR p-value:                    0.3385
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         1.9174      0.547      3.504      0.000       0.845       2.990
Focus[T.They]    -0.

**Interactive Model**

**E14**

escape_L2 ~ Focus * Agency

In [33]:
E14 = smf.logit(
    "escape_L2 ~ Focus * Agent",
    data=no_goal
    ).fit()

print(E14.summary())

Optimization terminated successfully.
         Current function value: 0.457646
         Iterations 6
                           Logit Regression Results                           
Dep. Variable:              escape_L2   No. Observations:                   66
Model:                          Logit   Df Residuals:                       62
Method:                           MLE   Df Model:                            3
Date:                Tue, 21 Jul 2026   Pseudo R-squ.:                 0.03479
Time:                        16:41:07   Log-Likelihood:                -30.205
converged:                       True   LL-Null:                       -31.293
Covariance Type:            nonrobust   LLR p-value:                    0.5365
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               1.9459      0.617      3.153      0.002       0.736       3.156
Fo

While goal participants seem to produce a coherent behaviourla structure, the no_goal participants don't. They don't seem to have the same behaviour constraints. 

4. Model Comparisons

In [34]:
from scipy.stats import chi2

def lr_test(model_small, model_large):

    LR = 2 * (model_large.llf - model_small.llf)

    df = model_large.df_model - model_small.df_model

    p = chi2.sf(LR, df)

    return pd.Series({
        "LR": LR,
        "df": df,
        "p_value": p
    })

In [35]:
lr_test(E01, E03)

LR        -0.456672
df         0.000000
p_value         NaN
dtype: float64

In [36]:
lr_test(E01, E04)

LR         11.019634
df          1.000000
p_value     0.000902
dtype: float64

5. Summary Table

In [37]:
goal_behaviour = ["Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal"]

model_name = ["E01", "E02", "E03", "E04", "E04", "E05", "E05", "E06", "E06", "E07", "E07", "E07", "E08", "E08", "E08", "E09", "E09", "E09", "E10", "E10", "E10", "E10", "E11", "E12", "E13", "E13", "E14", "E14", "E14"]

outcome = ["Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2", "Escape_L2"]

formula = ["escape_L2 ~ Focus", "escape_L2 ~ Agent", "escape_L2 ~ Goal_Type", "escape_L2 ~ Focus + Agent","escape_L2 ~ Focus + Agent", "escape_L2 ~ Focus + Goal_Type", "escape_L2 ~ Focus + Goal_Type", "escape_L2 ~ Agent + Goal_Type", "escape_L2 ~ Agent + Goal_Type", "escape_L2 ~ Focus * Agent", "escape_L2 ~ Focus * Agent", "escape_L2 ~ Focus * Agent", "escape_L2 ~ Focus * Goal_Type", "escape_L2 ~ Focus * Goal_Type", "escape_L2 ~ Focus * Goal_Type", "escape_L2 ~ Agent * Goal_Type", "escape_L2 ~ Agent * Goal_Type", "escape_L2 ~ Agent * Goal_Type", "escape_L2 ~ Focus * Agent + Goal_Type", "escape_L2 ~ Focus * Agent + Goal_Type", "escape_L2 ~ Focus * Agent + Goal_Type", "escape_L2 ~ Focus * Agent + Goal_Type", "escape_L2 ~ Focus", "escape_L2 ~ Agent", "escape_L2 ~ Focus + Agent", "escape_L2 ~ Focus + Agent", "escape_L2 ~ Focus * Agent", "escape_L2 ~ Focus * Agent", "escape_L2 ~ Focus * Agent" ]

predictor= ["They", "Agent", "goal_non_frequent", "They", "Agent", "They", "goal_non_frequent", "goal_non_frequent", "Agent", "They", "Agent", "They:Agent", "They", "goal_non_frequent", "They:goal_non_frequent", "goal_non_frequent", "Agent", "Agent:goal_non_frequent", "They", "goal_non_frequent", "Agent", "They:Agent", "They", "Agent", "They", "Agent", "They", "Agent", "They:Agent"]

coef = [0.6943, 1.7327, 0.5833, 0.8763, 1.8410, 0.7032, 0.5929, 0.6242, 1.7527, 0.7282, 1.5755, 19.7294, 0.1054, 0.1382, 1.8768, 0.6061, 1.7047, 0.1315, 0.7424, 0.6405, 1.5997, 18.5474, -0.9474, 0.1625, -0.9409 , 0.0723,  -0.9904, -3.812e-16, 0.1431]

std_error = [0.484, 0.645, 0.448, 0.501, 0.652, 0.487, 0.451, 0.463, 0.648, 0.521, 0.679, 2.15e+04, 0.592, 0.526, 1.238, 0.499, 0.808, 1.356, 0.526, 0.469, 0.684, 1.19e+04, 0.651, 0.674, 0.654, 0.688, 0.811, 0.976, 1.377]

coef_p = [0.151, 0.007, 0.193, 0.080, 0.005, 0.149, 0.189, 0.178, 0.007, 0.162, 0.020, 0.999, 0.859, 0.793, 0.130, 0.224, 0.035, 0.923, 0.158, 0.172, 0.019,0.999, 0.146, 0.809, 0.150, 0.916, 0.222, 1.000, 0.917]

LLR_p = [0.1386, 0.001628, 0.1875, 0.001351, 0.001351, 0.1381, 0.1381, 0.002758, 0.002758, 0.002293, 0.002293, 0.002293, 0.07919, 0.07919, 0.07919, 0.008117, 0.008117, 0.008117, 0.002514, 0.002514, 0.002514, 0.002514, 0.1421, 0.8084, 0.3385, 0.3385, 0.5365, 0.5365, 0.5365]

observations= [132, 132, 132, 132, 132, 132, 132, 132, 132, 132, 132,132, 132, 132, 132, 132, 132, 132, 132, 132, 132, 132, 66, 66, 66, 66, 66, 66, 66]

In [38]:
goal_split_escape_summary = pd.DataFrame({
    "Goal_Behaviour": goal_behaviour,
    "Model": model_name,
    "Outcome": outcome,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Std_err": std_error,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": observations
})

goal_split_escape_summary

,Goal_Behaviour,Model,Outcome,Formula,Predictor,Coef,Std_err,Coef_p,Model_LLR_p,N
0,Goal,E01,Escape_L2,escape_L2 ~ Focus,They,6.943000e-01,0.484,0.151,0.138600,132
1,Goal,E02,Escape_L2,escape_L2 ~ Agent,Agent,1.732700e+00,0.645,0.007,0.001628,132
2,Goal,E03,Escape_L2,escape_L2 ~ Goal_Type,goal_non_frequent,5.833000e-01,0.448,0.193,0.187500,132
3,Goal,E04,Escape_L2,escape_L2 ~ Focus + Agent,They,8.763000e-01,0.501,0.080,0.001351,132
4,Goal,E04,Escape_L2,escape_L2 ~ Focus + Agent,Agent,1.841000e+00,0.652,0.005,0.001351,132
5,Goal,E05,Escape_L2,escape_L2 ~ Focus + Goal_Type,They,7.032000e-01,0.487,0.149,0.138100,132
6,Goal,E05,Escape_L2,escape_L2 ~ Focus + Goal_Type,goal_non_frequent,5.929000e-01,0.451,0.189,0.138100,132
7,Goal,E06,Escape_L2,escape_L2 ~ Agent + Goal_Type,goal_non_frequent,6.242000e-01,0.463,0.178,0.002758,132
8,Goal,E06,Escape_L2,escape_L2 ~ Agent + Goal_Type,Agent,1.752700e+00,0.648,0.007,0.002758,132
9,Goal,E07,Escape_L2,escape_L2 ~ Focus * Agent,They,7.282000e-01,0.521,0.162,0.002293,132


- Agency predicts escaping alternative L2 responses among participants who actually have a goal to protect. Thus, we could ask if ,once there isn't a goal scenario, what could agency help or contribute to. 

- Goal Type consistently exhibited positive coefficients across additive models, although its independent contribution remained modest and did not reach conventional levels of statistical significance.

- Given results of the full additive model, escape responses are best understood as the joint product of agency, perspective, and goal structure, although agency provides the strongest independent contribution. Adding Goal Type preserves the pattern of effects and yields a coherent cognitive account in which agency remains the dominant predictor while focus and goal structure contribute more modestly.